### PR TITLE
[For Review Purpose Only-Do not Merge] First Draft for Glibc Importer

### PR DESCRIPTION
**Pull request by harsh098**
This PR addresses #1362.
Parser has been added in order to review the parser for GLIBC Importer.
@TG1999 and @pombredanne please have a look.

### CODE REVIEW

File: vulnerabilities/importers/glibc.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    def advisory_data(self) -> Iterable[AdvisoryData]:
+        raise NotImplementedError
+
+
+def parse_advisory_data(glibc_advisory, file_name) -> AdvisoryData:
+    """
+    Takes in advisory data and returns a dictionary containing the useful parsed info.
+    """
+    content = glibc_advisory.split("\n")
+    subject = content[0]
```
Comment (TG1999) [inline vulnerabilities/importers/glibc.py:35]:
```suggestion
    if content:
        subject = content[0]
```

File: vulnerabilities/importers/glibc.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+                else max_affected_version
+            )
+
+    # TODO: Implement affected_packages = AffectedPackage(package=purl, VersionRange.from_versions())
+    # TODO: Implement to_advisory(advisory_dict)
+    advisory_dict["affected_packages"] = list({min_affected_version, max_affected_version})
+    return advisory_dict
+
+
+def sanitize_version(version: str):
```
Comment (TG1999) [inline vulnerabilities/importers/glibc.py:103]: Add docstring, and few doctests

File: vulnerabilities/importers/glibc.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    license_url = "https://sourceware.org/git/?p=glibc.git;a=blob_plain;f=LICENSES"
+    spdx_license_expression = "LGPL-2.1-only"
+
+    def advisory_data(self) -> Iterable[AdvisoryData]:
+        raise NotImplementedError
+
+
+def parse_advisory_data(glibc_advisory, file_name) -> AdvisoryData:
+    """
+    Takes in advisory data and returns a dictionary containing the useful parsed info.
```
Comment (TG1999) [inline vulnerabilities/importers/glibc.py:32]: Be assertive with docstrings.

File: vulnerabilities/importers/glibc.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    content = glibc_advisory.split("\n")
+    subject = content[0]
+    line_counter = 2
+    description = ""
+    date = ""
+    cve_id = ""
+    vulnerable_commits = []
+    fix_commits = []
+    for line in content[line_counter:]:
+        if line.strip() == "":
```
Comment (TG1999) [inline vulnerabilities/importers/glibc.py:43]:
```suggestion
        if not line.strip():
```

File: vulnerabilities/tests/test_glibc.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
+import os
```
Comment (TG1999) [inline vulnerabilities/tests/test_glibc.py:1]: Add headers for this file

File: vulnerabilities/importers/glibc.py [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+    repo_url = "git+https://sourceware.org/git/glibc.git"
+    license_url = "https://sourceware.org/git/?p=glibc.git;a=blob_plain;f=LICENSES"
+    spdx_license_expression = "LGPL-2.1-only"
+
+    def advisory_data(self) -> Iterable[AdvisoryData]:
+        raise NotImplementedError
+
+
+def parse_advisory_data(glibc_advisory, file_name) -> AdvisoryData:
+    """
```
Comment (TG1999) [inline vulnerabilities/importers/glibc.py:31]: Add a sample advisory that you are trying to parse here.

**Comment by harsh098** [comment]
Please Specify what version range should be passed

File: vulnerabilities/tests/test_glibc.py (hunk 5 above)
Comment (harsh098) [reply vulnerabilities/tests/test_glibc.py:1]: Please Elaborate on this one.

**Comment by harsh098** [comment]
First draft for importer published.

**Comment by TG1999** [comment]
@harsh098 closing this PR, feel free to open this once it's ready for review

**Pull request closed without merging**